lib-manual/helpers/webhook_verifier.rb

```diff
@@ -0,0 +1,74 @@
+require 'openssl'
+require 'securecompare'
+
+module MuxRuby
+  module Helpers
+    class WebhookVerifier
+      DEFAULT_TOLERANCE = 300
+      HEADER_SCHEMES = [:v1].freeze
+
+      # Initializer.
+      #
+      # @param [String] secret the webhook secret from your Mux control panel
+      # @param [Integer] tolerance the signature timing tolerance, in seconds (generally leave as is)
+      # @param [Array<Symbol>] header_schemes the list of accepted header schemes for this verifier
+      def initialize(secret: nil, tolerance: DEFAULT_TOLERANCE, header_schemes: [:v1])
+        raise "secret '#{secret.inspect}' must be a String" unless secret.is_a?(String)
+        raise "tolerance '#{tolerance.inspect}' must be a positive number." \
+          unless tolerance.is_a?(Integer) && tolerance > 0
+        raise "header schemes '#{header_schemes.inspect}' must all be in HEADER_SCHEMES: '#{HEADER_SCHEMES.inspect}'" \
+          unless header_schemes.all? { |h| HEADER_SCHEMES.include?(h) }
+            
+        @secret = secret.dup
+        @tolerance = tolerance
+        @header_schemes = header_schemes.map(&:to_s).sort.uniq.reverse
+      end
+
+      # Initializer.
+      #
+      # @param [String] request_body the raw, unmodified body of the request
+      # @param [String] header the Mux-Signature header
+      # @param [Time] current_timestamp (for test purposes) the current time expected for this webhook (defaults to `Time.utc`)
+      # @return [Boolean] true if webhook is verified; false otherwise
+      def verify(request_body:, header:, current_timestamp: Time.now.getutc)
```

Comment:
I don't remember this from Ruby land circa 2013, but is `arg:,` just specifying a nullish default value?

Reply:
It specifies a named argument rather than a positional argument. I think they're clearer.

Reply:
@mmcc this must have come after your Ruby time (it was right near the end of my Ruby time) -- keyword args, which are a _fantastic_ language feature:

https://thoughtbot.com/blog/ruby-2-keyword-arguments

In this case calling `verify` without `request_body:` and `header:` args would cause a runtime error.

In the past we would sometimes use an options hash and then have conditionals inside method to check for things that were passed into the options hash.

Reply:
Yeah, this just replaces the options hash.

@dylanjha , I didn't know you were a Ruby guy. Oh, do I have code for you to review...!